homeassistant/util/unit_system.py

```diff
@@ -118,16 +131,18 @@ def __init__(
         if errors:
             raise ValueError(errors)
 
-        self._name = name
-        self.accumulated_precipitation_unit = accumulated_precipitation
-        self.area_unit = area
-        self.length_unit = length
-        self.mass_unit = mass
-        self.pressure_unit = pressure
-        self.temperature_unit = temperature
-        self.volume_unit = volume
-        self.wind_speed_unit = wind_speed
-        self._conversions = conversions
+        object.__setattr__(self, "_name", name)
```

Comment:
After creating the PR I realized `super()` can also work and is cleaner:
```suggestion
        super().__setattr__("_name", name)
```
I did consider other options such as using the default init and doing the checks at `__post_init__` but it would need to make other changes as the input parameters name doesn't match the class stored variables